### DISCUSSION TITLE
[Assemble] Question : Taking only one pre-clone per cell

### DISCUSSION THREAD

**Discussion by gabrielathenes** [milaboratory/mixcr#1405] [Q&A]
Hello everyone, 

I am new at using Mixcr and I am trying to assemble clonotypes of a fastq file of antibody reads of the same VDJRegion for different cells. I have a cell barcode for each of these reads, and aligned it using this tag pattern. Only problem is I would like the Pre-clone assembler to keep only one pre-clone per cell, for example using a consensus algorithm that keeps the most frequent read for each cell. I tried this using the override parameter "-Passembler.maxConsensuses=1" but it does not seem to work.
 
Is there something I am missing ?

**Comment by mizraelson** [comment]
Hi, can you share the commands you use and the version of MiXCR?

**Reply by gabrielathenes** [reply]
Hi, thanks for your reply :

- This is the commandline : mixcr assemble alignments.vdjca clones.clna -Passembler.maxConsensuses=1 -a -s -j assemble_report.json --cell-level --assemble-clonotypes-by VDJRegion -f
- I did the alignment with this command : mixcr align processed_SRR8810651.fastq alignments.vdjca --preset rna-seq --species mmu --rna -t 20 -j alignment_report.json -f --tag-pattern ^(CELL:N{24})(R1:*)
- The mixcr version is the following :
MiXCR v4.5.0 (built Fri Sep 22 14:39:05 CEST 2023; rev=cdb24b4fb7; branch=no_branch; host=fv-az438-99)
RepSeq.IO v2.3.0 (rev=dcfd972ff0)
MiLib v3.3.0 (rev=50ba183fea)
Built-in V/D/J/C library: repseqio.v3.0.1

**Reply by gabrielathenes** [reply]
My fastq file looks like this. There is a cell barcode which consists of the first 24 characters, and the following ~350 characters are the VDJ region.
[example.txt](https://github.com/milaboratory/mixcr/files/13174770/example.txt)

**Reply by gabrielathenes** [reply]
I have about ~3000000 reads divided in  ~3000 cells, so I would like to have on consensus per cell to then use tree reconstruction algorithms

**Reply by mizraelson** [reply]
Hi, thanks for sharing the sample file. Is it only gonna be heavy chains in each cell or a light chain as well?

**Reply by gabrielathenes** [reply]
It's both sorry I forgot to mention

**Reply by gabrielathenes** [reply]
So for each cell I would like one consensus for light and one for heavy, the idea is to then use the findShmTrees methods to create trees without two nodes of the tree coming from the same cell

**Comment by mizraelson** [comment]
I see, that means that per cell you actually expect two consensuses at least. I would recommend to start with trying the following command.

```
mixcr analyze generic-ht-single-cell-amplicon \
    --species mmu \
    --rna \
    --assemble-clonotypes-by VDJRegion \
    --rigid-left-alignment-boundary \
    --rigid-right-alignment-boundary J \
    --tag-pattern "^(CELL:N{24})(R1:*)" \
      input.fastq.gz \
      result
```

This command will apply all the necessary parameters and filters  according to your library structure. It will also run [`mixcr groupClones`](https://mixcr.com/mixcr/reference/mixcr-groupClones/) which will group clones by CELL barcode leaving only the ones that are more abundant. 

Also, we are currently developing a dedicated algorithm for single cell linage trees reconstruction, which will be available  soon.
